tests/lib/nm/wired_test.py

```diff
@@ -113,3 +114,118 @@ def test_get_info_with_invalid_duplex(ethtool_mock, NM_mock):
         'mac-address': dev_mock.get_hw_address.return_value,
         'mtu': dev_mock.get_mtu.return_value
     }
+
+
+class TestWiredSetting(object):
+
+    def test_identity(self):
+        state = {}
+        obj1 = obj2 = nm.wired.WiredSetting(state)
+
+        assert obj1 == obj2
```

Comment:
Just wondering, how is it possible to create two objects where the first assert succeeds but the second one does not (or the other way round).

Reply:
Try to remove `__ne__` and you will find out :)
I `__eq__` has been implemented, usually `__ne__` will call it with a `not`. However, in PY2 it is not done by default (there is a need to implement `__ne__` explicitly) and in some special cases, one may implement the `__ne__` by not negating equality, but implementing something else which is "optimized".